pwiz_tools/Skyline/Alerts/AlertDlg.cs

```diff
@@ -44,21 +44,22 @@ public AlertDlg() : this(@"Alert dialog for Forms designer")
         {
         }
 
-        public AlertDlg(string message)
+        public AlertDlg(string message, string title = null)
         {
             InitializeComponent();
             _originalFormHeight = Height;
             _originalMessageHeight = labelMessage.Height;
             _labelPadding = messageScrollPanel.Width - labelMessage.MaximumSize.Width;
             Message = message;
             btnMoreInfo.Parent.Controls.Remove(btnMoreInfo);
-            Text = Program.Name;
+            Text = title ?? Program.Name;
             toolStrip1.Renderer = new NoBorderSystemRenderer();
         }
 
-        public AlertDlg(string message, MessageBoxButtons messageBoxButtons) : this(message)
+        public AlertDlg(string message, MessageBoxButtons messageBoxButtons, string title = null,
+            MessageBoxDefaultButton defaultButton = MessageBoxDefaultButton.Button1) : this(message, title)

```

Comment:
I know "MessageBoxDefaultButton" is the standard way of designating the default button in a MessageBox, but could we use "DialogResult" as the way to designate it in AlertDlg? The caller definitely knows which DialogResult they are intend to be the default button, but they might not be so sure about whether that's the first or second button.

Reply:
Excellent feedback, as usual. I'll look into these. 

I was going for minimal widespread code changes but it is perhaps just as well to make it clear that this doesn't behave exactly the way a MessageBox does. So yes, I should have swapped in MultiButtonMessageDlg where appropriate.

>> The advantage of doing things that way is that the constructors and other methods do not get cluttered with extra arguments that are only used in a few cases. 
If the objection is just that the ctor has a lot of default arguments, it's not a bother to create multiple ctors.

>> The disadvantage is that it takes several lines to show an AlertDlg if you are doing something nonstandard.
That seems like a fair sized disadvantage, especially when "nonstandard" encompasses things that are pretty normal for MessageBox use. They made the API for a reason, surely?